datahub-web-react/src/app/tags/TagsTable.tsx

```diff
@@ -0,0 +1,436 @@
+    }
+
+    const facets = facetData?.searchAcrossEntities?.facets || [];
+    const entityFacet = facets.find((facet) => facet?.field === 'entity');
+    const aggregations = entityFacet?.aggregations || [];
+
+    if (aggregations.length === 0) {
+        return <Text>Not applied</Text>;
+    }
+
+    // Get total count of entities this tag is applied to
+    const totalCount = aggregations.reduce((sum, agg) => sum + (agg?.count || 0), 0);
+
+    // Navigate to search with entity filter
+    const navigateToEntitySearch = (entityTypeValue: string) => {
+        // Convert the string to EntityType
+        const entityType = entityTypeValue as unknown as EntityType;
+
+        navigateToSearchUrl({
+            filters: [
+                ...entityFilters,
+                {
+                    field: 'entity',
+                    values: [entityType],
+                },
+            ],
+            unionType: UnionType.AND,
+            history,
+        });
+    };
+
+    // Navigate to search with just the tag filter
+    const navigateToAllTagSearch = () => {
+        navigateToSearchUrl({
+            filters: entityFilters,
+            unionType: UnionType.AND,
+            history,
+        });
+    };
+
+    return (
+        <CellContainer>
+            <div
+                style={{ cursor: 'pointer' }}
+                onClick={navigateToAllTagSearch}
+                role="button"
+                tabIndex={0}
+                onKeyDown={(e) => {
+                    if (e.key === 'Enter' || e.key === ' ') {
+                        navigateToAllTagSearch();
+                    }
+                }}
+            >
+                <Text style={{ color: colors.violet[500] }}>
+                    {totalCount} {totalCount === 1 ? 'entity' : 'entities'}
+                </Text>
+            </div>
+
+            {aggregations.slice(0, 3).map((agg) => {
+                if (!agg?.value || !agg?.count || agg.count === 0) return null;
+                return (
+                    <div
+                        key={agg.value}
+                        style={{ cursor: 'pointer' }}
+                        onClick={() => navigateToEntitySearch(agg.value)}
+                        role="button"
+                        tabIndex={0}
+                        onKeyDown={(e) => {
+                            if (e.key === 'Enter' || e.key === ' ') {
+                                navigateToEntitySearch(agg.value);
+                            }
+                        }}
+                    >
+                        <Text style={{ color: colors.violet[500] }}>
+                            {agg.count} {entityRegistry.getCollectionName(agg.value as unknown as EntityType)}
+                        </Text>
+                    </div>
+                );
+            })}
+        </CellContainer>
+    );
+});
+
+const TagsTable = ({ searchQuery, searchData, loading: propLoading, networkStatus, refetch }: Props) => {
```

Comment:
this file is huge with a ton going on. can we split it up? there's 3 different components and utility functions in here and each component has a ton of logic

Reply:
yeah i just took anna's code

Reply:
agree tho - i will probably make these changes next week

Reply:
refactored!